apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php

```diff
@@ -231,99 +231,6 @@ function load_wpcom_block_editor_nux() {
 }
 add_action( 'plugins_loaded', __NAMESPACE__ . '\load_wpcom_block_editor_nux' );
 
-/**
- * Re-register some core patterns to push them down in the inserter list.
- * The reason is that Dotcom curate the pattern list based on their look.
- */
-function reorder_curated_core_patterns() {
-	$pattern_names = array( 'core/social-links-shared-background-color' );
-	foreach ( $pattern_names as $pattern_name ) {
-		$pattern = \WP_Block_Patterns_Registry::get_instance()->get_registered( $pattern_name );
-		if ( $pattern ) {
-			unregister_block_pattern( $pattern_name );
-			register_block_pattern(
-				$pattern_name,
-				$pattern
-			);
-		}
-	}
-}
-
-/**
- * Return a function that loads and register block patterns from the API. This
- * function can be registered to the `rest_dispatch_request` filter.
- *
- * @param Function $register_patterns_func A function that when called will
- * register the relevant block patterns in the registry.
- */
-function register_patterns_on_api_request( $register_patterns_func ) {
-	/**
-	 * Load editing toolkit block patterns from the API.
-	 *
-	 * It will only register the patterns for certain allowed requests and
-	 * return early otherwise.
-	 *
-	 * @param mixed           $response
-	 * @param WP_REST_Request $request
-	 */
-	return function ( $response, $request ) use ( $register_patterns_func ) {
-		$route = $request->get_route();
-		// Matches either /wp/v2/sites/123/block-patterns/patterns or /wp/v2/block-patterns/patterns
-		// to handle the API format of both WordPress.com and WordPress core.
-		$request_allowed = preg_match( '/^\/wp\/v2\/(sites\/[0-9]+\/)?block\-patterns\/(patterns|categories)$/', $route );
-
-		if ( ! $request_allowed || ! apply_filters( 'a8c_enable_block_patterns_api', false ) ) {
-			return $response;
-		};
-
-		$register_patterns_func();
-
-		reorder_curated_core_patterns();
-
-		return $response;
-	};
-}
-add_filter(
-	'rest_dispatch_request',
-	register_patterns_on_api_request(
-		function () {
-			require_once __DIR__ . '/block-patterns/class-block-patterns-from-api.php';
-			( new Block_Patterns_From_API() )->register_patterns();
-		}
-	),
-	10,
-	2
-);
-
-/**
- * This function exists for back-compatibility. Patterns in Gutenberg v13.0.0 and
- * newer are loaded over the API instead of injected into the document. While we
- * register patterns in the API above, we still need to maintain compatibility
- * with older Gutenberg versions and WP 5.9 without Gutenberg.
- *
- * This can be removed once the rest API approach is in core WordPress and available
- * on Atomic and Simple sites. (Note that Atomic sites may disable the Gutenberg
- * plugin, so we cannot guarantee its availability.)
- *
- * @param object $current_screen An object describing the wp-admin screen properties.
- */
-function register_patterns_on_screen_load( $current_screen ) {
```

Comment:
@okmttdhr just realized that we also have to remove this unused code.

Reply:
![badge](https://img.shields.io/badge/review-ask-blue.svg)
@miksansegundo Do you mean we should do the same thing in https://github.com/Automattic/jetpack/pull/34162 as well?

Reply:
We have not migrated that part, so your question doesn't apply 🙂

Reply:
Sorry for being unclear!
My understanding is that `register_patterns_on_screen_load` is for back-compatibility. Without it, patterns may not be properly injected into sites running older versions of Gutenberg, as well as WordPress 5.9 without the Gutenberg plugin. 

Considering this, should we consider migrating the `register_patterns_on_screen_load` functionality to  https://github.com/Automattic/jetpack/pull/34162 to maintain support for these older versions?

Reply:
The comment on `register_patterns_on_screen_load` says:
```
This can be removed once the rest API approach is in core WordPress and available on Atomic and Simple sites. (Note that Atomic sites may disable the Gutenberg plugin, so we cannot guarantee its availability.)
```

My understanding is that we can remove it for these reasons:

- the block patterns REST API has been in WordPress core since version 6.0.0. See https://github.com/WordPress/wordpress-develop/commit/a807e86391817a45dd82f54bef234be1a28d4c6d
  - in other words, the block patterns REST API is not dependent on the Gutenberg plugin
- on simple and atomic sites, the WordPress version is auto-updated. The current WordPress version is 6.4.1, which integrates Gutenberg 16.2-16.7. 
  - in other words, there aren't any simple or atomic sites still using WordPress 5.9
   
Here is a reference to see the Gutenberg version included in WordPress core: https://developer.wordpress.org/block-editor/contributors/versions-in-wordpress/

Please let me know if I am mistaken. cc: @obenland @Automattic/lego

Reply:
Looks great to me! Thank you for the detailed explanation! 🤗